litr-demo/src/main/java/com/linkedin/android/litr/demo/RecordAudioFragment.kt

```diff
@@ -0,0 +1,93 @@
+package com.linkedin.android.litr.demo
+
+import android.Manifest
+import android.app.Activity
+import android.app.AlertDialog
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.linkedin.android.litr.MediaTransformer
+import com.linkedin.android.litr.demo.data.TargetMedia
+import com.linkedin.android.litr.demo.data.TransformationPresenter
+import com.linkedin.android.litr.demo.data.TransformationState
+import com.linkedin.android.litr.demo.databinding.FragmentAudioRecordBinding
+import com.linkedin.android.litr.io.AudioRecordMediaSource
+import com.linkedin.android.litr.utils.TransformationUtil
+import java.io.File
+
+class RecordAudioFragment : BaseTransformationFragment() {
+    private lateinit var binding: FragmentAudioRecordBinding
+
+    private lateinit var mediaTransformer: MediaTransformer
+    private var targetMedia: TargetMedia = TargetMedia()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mediaTransformer = MediaTransformer(context!!.applicationContext)
+
+        // This demo fragment requires Android M or newer, in order to support reading data from
+        // AudioRecord in a non-blocking way. Let's double check that the current device supports
+        // this.
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
```

Comment:
This requirement could be potentially removed in the future, but would require additional changes in `AudioRecordMediaSource`. We would need something like a ring buffer, with a separate thread reading via the older/blocking api and updating the buffer. Then we would read from this buffer in a non-blocking way. The downside is obviously additional complexity along with increased memory usage. For now, I thought requiring Android M or newer was better, at least for now.

Reply:
No need for anything that complicated, just leave things as is. Demo app is supposed to be a space for easy experimentation and sample code, not a prod ready app. What you are doing here is exactly what it is for.